ipaserver/plugins/cert.py

```diff
@@ -705,7 +705,8 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
             # fail if any email addr from DN does not appear in ldap entry
             email_addrs = csr_obj.subject.get_attributes_for_oid(
                     cryptography.x509.oid.NameOID.EMAIL_ADDRESS)
-            if len(set(email_addrs) - set(principal_obj.get('mail', []))) > 0:
+            if not _emails_are_valid(email_addrs,
+                                     principal_obj.get('mail', [])):
```

Comment:
This is checking the email addresses (if any) in the Subject DN.  We also need to check rfc822Names in the
Subject Alternative Name extension.